### PR TITLE
AAP-8257 EE Add Edit Form

### PR DESCRIPTION
**Pull request by prat98**
This PR is for [AAP-8257](https://issues.redhat.com/browse/AAP-8257) which encompasses the work around creating the add and edit forms on the new AWX UI.

### CODE REVIEW

File: frontend/awx/administration/execution-environments/ExecutionEnvironmentForm.tsx [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+import { requestGet, requestPatch, swrOptions } from '../../../common/crud/Data';
+import { useSearchParams } from '../../../../framework/components/useSearchParams';
+
+const PullOption = {
+  Always: 'Always pull container before running.',
+  Missing: 'Only pull the image if not present before running.',
+  Never: 'Never pull container before running.',
+};
+
+export interface IExecutionEnvInput {
```
Comment (kialam) [inline frontend/awx/administration/execution-environments/ExecutionEnvironmentForm.tsx:33]: Can we combine these two types into one?

**Review by kialam — APPROVED** [review]
pulled this down and tested locally. LGTM. I had one suggestion of combining the two interfaces for the form edit, however that's not a blocker for merging this work.

**Comment by kialam** [comment]
@prat98 we'll probably want to wait until https://github.com/ansible/ansible-ui/pull/1638 is merged before merging this one (it will need to be refactored to use the new orgs select in the edit EE form)

**Review by marshmalien — CHANGES_REQUESTED** [review]
It looks like we need to filter the registry credential input field by kind `/credentials/?credential_type__kind=registry`. In the old UI screenshot below, there are no credential results, but there are 3 credentials in the new UI screenshot.

_Side-by-side shot of new vs old UI_
Screenshot 2024-02-12 at 11 41 31 AM

Screenshot 2024-02-12 at 11 43 49 AM

**Review by marshmalien — APPROVED** [review]
lgtm!! 🎉

**Pull request merged**